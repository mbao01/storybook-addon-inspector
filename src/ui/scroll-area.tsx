"use client";

import * as React from "react";
import * as ScrollAreaPrimitive from "@radix-ui/react-scroll-area";

import { cn } from "../lib/utils";

const ScrollArea = React.forwardRef<
  React.ElementRef<typeof ScrollAreaPrimitive.Root>,
  React.ComponentPropsWithoutRef<typeof ScrollAreaPrimitive.Root>
>(({ className, children, ...props }, ref) => (
  <ScrollAreaPrimitive.Root
    ref={ref}
    className={cn("ia:relative ia:overflow-hidden", className)}
    {...props}
  >
    <ScrollAreaPrimitive.Viewport className="ia:h-full ia:w-full ia:rounded-[inherit]">
      {children}
    </ScrollAreaPrimitive.Viewport>
    <ScrollBar />
    <ScrollAreaPrimitive.Corner />
  </ScrollAreaPrimitive.Root>
));
ScrollArea.displayName = ScrollAreaPrimitive.Root.displayName;

const ScrollBar = React.forwardRef<
  React.ElementRef<typeof ScrollAreaPrimitive.ScrollAreaScrollbar>,
  React.ComponentPropsWithoutRef<typeof ScrollAreaPrimitive.ScrollAreaScrollbar>
>(({ className, orientation = "vertical", ...props }, ref) => (
  <ScrollAreaPrimitive.ScrollAreaScrollbar
    ref={ref}
    orientation={orientation}
    className={cn(
      "ia:flex ia:touch-none ia:select-none ia:transition-colors ia:bg-gray-50 ia:hover:bg-gray-100 ia:dark:bg-gray-50/10 ia:dark:hover:bg-gray-100/20",
      orientation === "vertical" &&
        "ia:h-full ia:w-2.5 ia:border-l ia:border-l-transparent ia:p-[1px]",
      orientation === "horizontal" &&
        "ia:h-2.5 ia:flex-col ia:border-t ia:border-t-transparent ia:p-[1px]",
      className,
    )}
    {...props}
  >
    <ScrollAreaPrimitive.ScrollAreaThumb className="ia:relative ia:flex-1 ia:rounded-full ia:bg-gray-400 ia:dark:bg-muted" />
  </ScrollAreaPrimitive.ScrollAreaScrollbar>
));
ScrollBar.displayName = ScrollAreaPrimitive.ScrollAreaScrollbar.displayName;

export { ScrollArea, ScrollBar };
