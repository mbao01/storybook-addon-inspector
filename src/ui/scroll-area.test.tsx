import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import { ScrollArea, ScrollBar } from "./scroll-area";

describe("ScrollArea", () => {
  it("renders scroll area with children", () => {
    render(
      <ScrollArea>
        <div>Content</div>
      </ScrollArea>,
    );

    const content = screen.getByText("Content");
    expect(content).toBeInTheDocument();
  });

  it("forwards ref to scroll area element", () => {
    const ref = { current: null };
    render(
      <ScrollArea ref={ref}>
        <div>Content</div>
      </ScrollArea>,
    );

    expect(ref.current).toBeInstanceOf(HTMLDivElement);
  });
});

describe("ScrollBar", () => {
  it("renders scroll bar with vertical orientation by default", () => {
    const { asFragment } = render(
      <ScrollArea>
        <ScrollBar orientation="vertical" />
      </ScrollArea>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders scroll bar with vertical orientation", () => {
    const { asFragment } = render(
      <ScrollArea>
        <ScrollBar orientation="vertical" />
      </ScrollArea>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders scroll bar with horizontal orientation", () => {
    const { asFragment } = render(
      <ScrollArea>
        <ScrollBar orientation="horizontal" />
      </ScrollArea>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
