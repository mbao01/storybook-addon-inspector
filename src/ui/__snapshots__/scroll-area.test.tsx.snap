// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ScrollBar > renders scroll bar with vertical orientation by default 1`] = `
<DocumentFragment>
  <div
    class="ia:relative ia:overflow-hidden"
    dir="ltr"
    style="position: relative; --radix-scroll-area-corner-width: 0px; --radix-scroll-area-corner-height: 0px;"
  >
    <style>
      [data-radix-scroll-area-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-scroll-area-viewport]::-webkit-scrollbar{display:none}
    </style>
    <div
      class="ia:h-full ia:w-full ia:rounded-[inherit]"
      data-radix-scroll-area-viewport=""
      style="overflow-x: hidden; overflow-y: scroll;"
    >
      <div
        style="min-width: 100%; display: table;"
      />
    </div>
  </div>
</DocumentFragment>
`;
